### PR TITLE
[ch3751] Add test to confirm we can donate with a subscription running

### PR DESCRIPTION
**Pull request by cacahouwete**
https://app.clubhouse.io/enmarche/story/3751/emp%C3%AAcher-les-personnes-qui-font-d%C3%A9j%C3%A0-un-don-mensuel-d-en-faire-un-second

### CODE REVIEW

File: src/Donation/DonationRequestUtils.php [hunk 1]
```diff
@@ -88,6 +91,11 @@ public function createFromRequest(Request $request, ?Adherent $currentUser): Don
         return $donation;
     }
 
+    public function getDuration(Request $request): int
+    {
+        return (int) $request->query->getInt('abonnement', PayboxPaymentSubscription::NONE);
```
Comment (ottaviano) [inline src/Donation/DonationRequestUtils.php]: Cast is useless, getInt returns only integer value

Comment (cacahouwete) [reply src/Donation/DonationRequestUtils.php]: Correct I did it without thinking :s

File: src/Form/DonationRequestType.php [hunk 2]
```diff
@@ -104,15 +100,15 @@ public function buildView(FormView $view, FormInterface $form, array $options)
 
     public function createInitialDonationRequest(FormEvent $formEvent): void
     {
-        if (!$formEvent->getData()) {
+        if (!$donationRequest = $formEvent->getData()) {
```
[the remark this answers is not in the document]
Comment (cacahouwete) [reply src/Form/DonationRequestType.php]: It was used in a previous thing I did before run the pr :s, forgot to remove it :s

File: features/donation.feature [hunk 3]
```diff
@@ -153,6 +153,7 @@ Feature: The goal is to donate one time or multiple time with a subscription
     When I click on the "1" "img" element
     Then I should see "Votre soutien compte beaucoup pour moi."
 
+#    Check if I can't continue create a new subscription and then can cancel a subscription
```
Comment (thomasbisignani) [inline features/donation.feature]: I'm don't really agree with this comment indentation. Whitespaces are useless here. 

Please juste write `# Check if I can't continue... ` indented with the `Given I am on "/don"`

File: features/donation.feature [hunk 4]
```diff
@@ -196,3 +198,44 @@ Feature: The goal is to donate one time or multiple time with a subscription
...
+    And I should see "Vous faites déjà un don mensuel à La République En Marche ! Vous pouvez vous rendre sur votre profil pour l’annuler ou faire un nouveau don unique."
+
+    When I follow "faire un nouveau don unique"
+    Then I should be on "/don/coordonnees?montant=50"
+    And I should not see "200€ / mois"
+
+    When I click the "donation_check_label" element
+    And I click the "donation_check_nationality_label" element
+    And I press "Je donne"
+
```
Comment (B-Galati) [inline features/donation.feature]: No newline here

**Pull request merged**